### PR TITLE
Remove obsolete cylc-6 back-compat code.

### PR DESCRIPTION
**Pull request by hjoliver**
This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (just removes an obsolete back-compat check).
- [x] No change log entry required (just removes ...ditto...).
- [x] No documentation update required.

### CODE REVIEW

**Comment by hjoliver** [comment]
(I guess we missed this bit when cleaning out the cruft).

File: cylc/flow/config.py [hunk 1]
```diff
@@ -1618,10 +1615,6 @@ def generate_taskdefs(self, orig_expr, left_nodes, right, seq):
             for item in self.cfg['runtime'][name]['outputs'].items():
                 if item not in taskdef.outputs:
```
Comment (dwsutherland) [inline cylc/flow/config.py]: @trwhitcomb, when you change the following to a set operation,  you won't need the `if` statement after this change.

Comment (hjoliver) [reply cylc/flow/config.py]: Could you clarify that statement @dwsutherland ?  ("won't need the if statement after this change" ... do you mean the if state *in* this change, or the one *before* this change?  - if the latter, maybe you didn't see this comment: https://github.com/cylc/cylc-flow/pull/3259#discussion_r310430133 )

Comment (hjoliver) [reply cylc/flow/config.py]: @trwhitcomb can safely ignore this change, in his PR.

Comment (dwsutherland) [reply cylc/flow/config.py]: @hjoliver : https://github.com/cylc/cylc-flow/pull/3268/files#diff-6f57d0a14add56d573847aef0ac65984

Comment (dwsutherland) [reply cylc/flow/config.py]: Yes, that's the one.

Comment (hjoliver) [reply cylc/flow/config.py]: (understood now!)

**Comment by hjoliver** [comment]
Travis failure was genuine. There was a functional test for this back-compat.  (Just removed).

**Comment by hjoliver** [comment]
And rebased.

**Comment by kinow** [comment]
Three approval reviews, so merging :+1:

**Pull request merged**